src/logic/calculate.js

```diff
@@ -0,0 +1,98 @@
+import { evaluate } from './operate';
+import ACTIONS from './actions';
+
+function reducer(state, { type, payload }) {
+  switch (type) {
+    case ACTIONS.ADD_DIGIT:
+      if (state.overwrite) {
+        return {
+          ...state,
+          currentOperand: payload.digit,
+          overwrite: false,
+        };
+      }
+      if (payload.digit === '0' && state.currentOperand === '0') {
+        return state;
+      }
+      if (payload.digit === '.' && state.currentOperand.includes('.')) {
+        return state;
+      }
+
+      return {
+        ...state,
+        currentOperand: `${state.currentOperand || ''}${payload.digit}`,
+      };
+    case ACTIONS.CHOOSE_OPERATION:
+      if (state.currentOperand == null && state.previousOperand == null) {
+        return state;
+      }
+
+      if (state.currentOperand == null) {
+        return {
+          ...state,
+          operation: payload.operation,
+        };
+      }
+
+      if (state.previousOperand == null) {
+        return {
+          ...state,
+          operation: payload.operation,
+          previousOperand: state.currentOperand,
+          currentOperand: null,
+        };
+      }
+
+      return {
+        ...state,
+        previousOperand: evaluate(state),
+        operation: payload.operation,
+        currentOperand: null,
+      };
+
+    case ACTIONS.CLEAR:
+      return {};
+
+    case ACTIONS.MAKE_NEGATIVE:
+      if (state.currentOperand !== null) {
+        const newOperand = parseFloat(state.currentOperand) * -1;
+        return {
+          ...state,
+          currentOperand: newOperand.toString(),
+        };
+      }
+      return state;
+
+    case ACTIONS.GIVE_PERCENTAGE:
+      if (state.currentOperand !== null) {
+        const newOperand = (parseFloat(state.currentOperand) / 100).toString();
+        return {
+          ...state,
+          currentOperand: newOperand,
+        };
+      }
+      return state;
+
+    case ACTIONS.EVALUATE:
+      if (
+        state.operation == null
+        || state.currentOperand == null
+        || state.previousOperand == null
+      ) {
+        return state;
+      }
+
+      return {
+        ...state,
+        overwrite: true,
+        previousOperand: null,
+        operation: null,
+        currentOperand: evaluate(state),
+      };
+
+    default:
+      return state;
+  }
+}
+
```

Comment:
- [ ] 👋 Hello there! First of all, awesome job on the work you've done so far with calculate.js! 🎉 But, it looks like there's a small adjustment needed to align with the project requirements. 

1. 📥 Please take a moment to copy the specific files required for this project:

  - calculate.js from [this link](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js)

  - operate.js from [this link](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js)

2. 🗑️ Next, in your current calculate.js and operate.js files, kindly delete their existing content.

3. 🔄 Replace the content you've deleted with the code you downloaded from the Microverse links. 📄

4. 🧐 The reason for this change is to ensure compliance with the project requirements 📋, which specify using these specific files provided by Microverse. This will help keep everything aligned with the project's objectives. 🎯

By following these steps, you'll be spot on with the project's requirements and can concentrate on the React events aspect, which is super cool! If you have any questions or need more help, don't hesitate to reach out. 🚀 Good luck with your project! 🌟